File: node/http/controllers/publishController.js
import express from "express";
import { v4 as uuidv4 } from 'uuid';
import { axiosInstance } from "../utils/axios.js";

const router = express.Router();

export const publishController = router.get("/", async (_req, res) => {
    try {
        const orderId = uuidv4()
        const data = [
            { "orderId": orderId }
        ]
        const response = await axiosInstance.post("/orders", data)
        console.log(response.status ?? "no status code returned")
        res.json({ "msg": `Order published with order id: ${orderId}` })
    } catch (error) {
        console.error(error)
        res.json({ "err": error })
    }
})

File: go/http/controllers/publishController.go
package controllers

import (
	"bytes"
	"encoding/json"
	"fmt"
	"log"
	"net/http"

	"github.com/gofiber/fiber/v2"
	"github.com/google/uuid"
)

type Order struct {
	OrderId string `json:"orderId"`
}

func PublishController(cxt *fiber.Ctx) error {
	uuid := uuid.New()

	orders := []Order{
		{
			OrderId: uuid.String(),
		},
	}
	// Marshal the orders struct to JSON
	t, err := json.Marshal(orders)
	res, err := http.Post("http://localhost:3500/v1.0/publish/pubsub/orders", "application/json", bytes.NewBuffer(t))

	fmt.Println(res)

	if err != nil {
		log.Fatal(err)
	}

	defer res.Body.Close()
	return cxt.JSON(fiber.Map{"msg": "Order published with order id: " + uuid.String()})
}
